pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -642,12 +642,26 @@ func getJobName(cj *batchv1.CronJob, scheduledTime time.Time) string {
 }
 
 // nextScheduledTimeDuration returns the time duration to requeue based on
-// the schedule and current time. It adds a 100ms padding to the next requeue to account
+// the schedule and last schedule time. It adds a 100ms padding to the next requeue to account
 // for Network Time Protocol(NTP) time skews. If the time drifts are adjusted which in most
 // realistic cases would be around 100s, scheduled cron will still be executed without missing
 // the schedule.
-func nextScheduledTimeDuration(sched cron.Schedule, now time.Time) *time.Duration {
-	t := sched.Next(now).Add(nextScheduleDelta).Sub(now)
+func nextScheduledTimeDuration(cj batchv1.CronJob, sched cron.Schedule, now time.Time) *time.Duration {
```

Comment:
I've noticed there's a quite a lot of similarities between this method and `getNextScheduleTime`, I'll followup with a squash. But this is very good - thank you!

Reply:
Yeah that's true, was trying to keep changes local but that had its cons.
Thank you!